File: Java/Stack/ReverseString.java
package Java.Stack;

import java.util.Stack;

public class ReverseString {
    public static void main(String[] args) {
        String s = "hello";
        System.out.println("Original: " + s);
        System.out.println("Reversed: " + reverseString(s));
    }

    public static StringBuilder reverseString(String str){
        Stack<Character> stack = new Stack<>();
        for (char c : str.toCharArray())
        {
            stack.push(c);
        }        
        StringBuilder reversed = new StringBuilder();
        while (!stack.isEmpty()) {
            reversed.append(stack.pop());

        }

        return reversed;
    }
}

File: Swift/Stack/ReverseString.swift
// Reverse a String using Stack
// Problem

// Given a string s, reverse it using a stack.

// Example

// Input:  "hello"
// Output: "olleh"

func reverseString(_ s: String) -> String{
    var stack: [Character] = []
    //pushing each character in stack
    for char in s {
        stack.append(char)
    }
    var reversed = ""
    //pop characters to build reversed string
    while !stack.isEmpty{
        reversed.append(stack.removeLast())

    }
    return reversed
}

let input = "hello"
print("Original: \(input)")
print("Reversed: \(reverseString(input))")